Submitted upstream.

From 4c3eb19a11dfe5c88c902481a8294c4f675fcd03 Mon Sep 17 00:00:00 2001
From: "Jan (janneke) Nieuwenhuizen" <janneke@gnu.org>
Date: Sun, 19 Apr 2020 23:33:28 +0200
Subject: [PATCH] Support cross-compiling to the Hurd.

* cnf/hints/gnu: New file.
* cnf/configure_tool.sh: Guess it.
---
 cnf/configure_tool.sh |  4 ++++
 cnf/hints/gnu         | 21 +++++++++++++++++++++
 2 files changed, 25 insertions(+)
 create mode 100644 cnf/hints/gnu

diff --git a/cnf/configure_tool.sh b/cnf/configure_tool.sh
index 32201c0..7b0baa4 100644
--- a/cnf/configure_tool.sh
+++ b/cnf/configure_tool.sh
@@ -266,6 +266,10 @@ if not hinted 'osname'; then
 			define osname "bsd"
 			result "BSD"
 			;;
+		*-gnu*)
+			define osname "gnu"
+			result "GNU"
+			;;
 		*)
 			result "no"
 			;;
diff --git a/cnf/hints/gnu b/cnf/hints/gnu
new file mode 100644
index 0000000..a0583a2
--- /dev/null
+++ b/cnf/hints/gnu
@@ -0,0 +1,21 @@
+# Hurd syscalls
+d_voidsig='define'
+d_nanosleep='undef'
+d_clock_gettime='define'
+d_clock_getres='define'
+d_clock_nanosleep='define'
+d_clock='define'
+
+# From the original linux.sh
+usemallocwrap='define'
+
+# libraries to test
+libswanted='m crypt pthread nm ndbm gdbm dbm db dl gdbm_compat'
+
+d_procselfexe='undef'
+procselfexe='"undef"'
+
+st_ino_sign=1
+st_ino_size=8
+
+d_fcntl_can_lock='define'
-- 
2.26.0

